analyses/DESeq_analysis/run-DESeq-analysis.R

```diff
@@ -0,0 +1,200 @@
+    resOrdered <- res[order(rownames(res)),]
+
+    Result <- resOrdered
+
+
+#Save subset of table with samples representing the histology in the DEG comparison to a variable.
+HIST_sample_type_df_filtered <- sample_type_df_filtered[which(sample_type_df_filtered$Type %in% c(histology_filtered[I])),]
+ 
+#Define study ID as all cohorts represented by the pateints involved in DEG comparison
+STUDY_ID <- paste(unique(hist$cohort[which(hist$Kids_First_Biospecimen_ID %in% HIST_sample_type_df_filtered$Case_ID)]),collapse=";",sep=";")
+
+#Record number of samples represnt the GTEX tissue used in the comparison
+GTEX_Hits <- length(sample_type_df_filtered[which(sample_type_df_filtered$Type %in% c(GTEX_filtered[J])),1])
+
+#Determine the mean TPM of the tissue. If there are multiple samples use the mean TPM
+if(GTEX_Hits > 1) GTEX_MEAN_TPMs = round(apply(TPMData_filtered[match(rownames(Result),rownames(TPMData_filtered)),which(colnames(TPMData_filtered) %in%  sample_type_df_filtered[which(sample_type_df_filtered$Type %in% c(GTEX_filtered[J])),1])]  ,MARGIN=1, mean ),2)
+
+#Determine the mean TPM of the tissue. If there is one sample just use the single TPM value of the sample
+if(GTEX_Hits <= 1) GTEX_MEAN_TPMs = TPMData_filtered[match(rownames(Result),rownames(TPMData_filtered)),which(colnames(TPMData_filtered) %in%  sample_type_df_filtered[which(sample_type_df_filtered$Type %in% c(GTEX_filtered[J])),1])]
+
+#Record number of samples represnt the cancer patient data used in the comparison
+Cancer.Hist_Hits <- length(sample_type_df_filtered[which(sample_type_df_filtered$Type %in% c(histology_filtered[I])),1])
+
+#Determine the mean TPM of the tissue. If there are multiple samples use the mean TPM
+if(Cancer.Hist_Hits > 1) Histology_MEAN_TPMs <- round(apply(TPMData_filtered[match(rownames(Result),rownames(TPMData_filtered)),which(colnames(TPMData_filtered) %in%  sample_type_df_filtered[which(sample_type_df_filtered$Type %in% c(histology_filtered[I])),1])]  ,MARGIN=1, mean ),2)
+
+#Determine the mean TPM of the tissue. If there is one sample just use the single TPM value of the sample
+if(Cancer.Hist_Hits <= 1) Histology_MEAN_TPMs <- TPMData_filtered[match(rownames(Result),rownames(TPMData_filtered)),which(colnames(TPMData_filtered) %in%  sample_type_df_filtered[which(sample_type_df_filtered$Type %in% c(histology_filtered[I])),1])]
+
+#Round the mean TPMs to the 2 decimal places
+Histology_MEAN_TPMs <- round(Histology_MEAN_TPMs,2)
+GTEX_MEAN_TPMs <- round(GTEX_MEAN_TPMs,2)
+
+
+#Create Final Dataframe with all the info calculated and extracted from histology file Including EFO/MONDO codes where available and RMTL status
+Final_Data_Table <- data.frame(
+  datasourceId <- paste(strsplit(histology_filtered[I],split="_")[[1]][1],"vs_GTex",sep="_"),
+  datatypeId <- "rna_expression",
+  cohort <- paste(unique(hist$cohort[which(hist$Kids_First_Biospecimen_ID %in% HIST_sample_type_df_filtered$Case_ID)]),collapse=";",sep=";"),
+  gene_symbol <- rownames(Result),
```

Comment:
```suggestion
  Gene_symbol <- rownames(Result),
```
for compatibility with the annotator

Reply:
Done.